src/sleep/sleep.c

```diff
@@ -298,6 +293,8 @@ static int custom_timer_suspend(const SleepConfig *sleep_config) {
                 else
                         suspend_interval = total_suspend_interval;
 
+                before_timestamp = now(CLOCK_BOOTTIME);
```

Comment:
Maybe move the declaration here? With this long list of variables it's hard to figure out where they become valid.